pkg/guacrest/server/retrieveDependencies.go

```diff
@@ -288,10 +304,9 @@ func neighborToNode(ctx context.Context, neighborsNode gql.NeighborsNeighborsNod
 	return nil, helpers.Err500
 }
 
-// Maps nodes in the input to purls, ignoring nodes that are not package version
-// nodes.
+// Maps nodes in the input to a map of "pkg id : purl", ignoring nodes that are not package version nodes.
 func mapPkgNodesToPurls(ctx context.Context, gqlClient graphql.Client,
-	nodes []node) ([]string, error) {
+	nodes []node) (map[string]string, error) {
```

Comment:
Is there a reason for this change? I don't see the ID's used anywhere.

Reply:
This is needed for the follow up PR, so I have just done the changes here along with the rest of the dependency retrieval.